Bug: `File exists: '/home/runner/.cache/molbloom'`

Seen in CI logs with `molbloom==2.3.4` and Python 3.11:

```none
.venv/lib/python3.11/site-packages/molbloom/__init__.py:103: in buy
    _load_filter(catalog)
.venv/lib/python3.11/site-packages/molbloom/__init__.py:66: in _load_filter
    _load_big_filter(name)
.venv/lib/python3.11/site-packages/molbloom/__init__.py:43: in _load_big_filter
    os.makedirs(_DEFAULT_PATH)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = '/home/runner/.cache/molbloom', mode = 511, exist_ok = False

>   ???
E   FileExistsError: [Errno 17] File exists: '/home/runner/.cache/molbloom'
```

Looks like it should be `os.makedirs(_DEFAULT_PATH, exist_ok=True)`

--- a/molbloom/__init__.py
+++ b/molbloom/__init__.py
@@ -39,8 +39,7 @@ def _download_progress(count, block_size, total_size):
 def _load_big_filter(name):
     # check if it's present
     filter_path = os.path.join(_DEFAULT_PATH, f"{name}.bloom")
-    if not os.path.exists(_DEFAULT_PATH):
-        os.makedirs(_DEFAULT_PATH)
+    os.makedirs(_DEFAULT_PATH, exist_ok=True)
     if not os.path.exists(filter_path):
         print(f"Starting {name} download to cache directory {_DEFAULT_PATH}")
         import urllib.request

